__tests__/mock-xhr-test.ts

```diff
@@ -0,0 +1,93 @@
+jest.unmock("node-fetch");
+
+import fetch from "node-fetch";
+import createServer from "./helpers/createServer";
+import MockServerController from "./helpers/mockServerController";
+
+function xhr(port, options = {}) {
+  return fetch(`http://localhost:${port}`, {
+    ...options,
+    headers: {
+      ...options.headers,
+      "content-type": "application/json"
+    }
+  });
+}
+
+describe("Mock - XHR", () => {
+  let s;
```

Comment:
I understand your concern, the problem is that the thing s returns is hard to name, because it's a shitty abstraction. It is a shitty abstraction because it's a test helper that needs to expose a lot of "unrelated" information to bring any value. I could name it `serverController` but that is not really what it is. I would love to deconstruct it right away, but afaik that just doesn't work if I need to define the block scope on the outer describe.

Reply:
To me it's not `createServer` but rather `createTestContext` or `createTestEnvironment`

Reply:
But that is also not entirely true, a test context would also include the mock server to be started, wouldn't it?

Reply:
Then we could encapsulate the whole thing inside the context.

Because right now we're creating a server with `createServer` and take its `port` to pass into a controller to start a controller to get another `port`

Reply:
That is for me completely reasonable, as one is a proxy target and the other is the mockserver implementation.

The problem is that we need to kill proxy target again, so I can't just pass the port.